docs/guide/overview.asciidoc

```diff
@@ -0,0 +1,51 @@
+[[overview]]
+== Overview
+
+Elastic APM is an application performance monitoring system built on the Elastic Stack.
+It allows you to monitor software services and applications in real time,
+collecting detailed performance information on response time for incoming requests,
+database queries,
+calls to caches,
+external HTTP requests,
+etc.
+This makes it easier to pinpoint and fix performance problems quickly.
+
+Elastic APM also collects automatically unhandled errors and exceptions.
+Errors are grouped based primarily on the stacktrace,
+so you can identify new errors as they appear and keep an eye on how many times specific errors happen. 
+
+NOTE: This guide will indiscriminately use the word service for both services and applications.
+
+[[components]]
+[float]
+=== APM Components
+
+Elastic APM consists of four components:
+
+* {ref}/index.html[Elasticsearch]
+* {apm-agents-ref}/index.html[APM agents]
+* {apm-server-ref}/index.html[APM Server]
+* {kibana-ref}/xpack-apm.html[Kibana APM UI]
+
+image::apm-architecture.png[Architecture of Elastic APM]
+
+*APM agents* are open source libraries written in the same language as your service.
+You install them into your service as you would install any other library.
+They instrument your code and collect performance data and errors at runtime.
+This data is buffered for a short period and sent on to APM Server.
+
+Out of the box,
+APM automatically instruments web frameworks,
```

Comment:
The "automatically out of the box" part comes from when we only had agents for dynamically typed languages, this is not true eg for Go. I suggest to remove that in a future iteration :)